src/sources/journald.rs

```diff
@@ -57,17 +61,25 @@ impl SourceConfig for JournaldConfig {
             })
             .collect::<HashSet<String>>();
 
-        Ok(journald_source(journal, out, units))
+        let checkpointer = Checkpointer::new(data_dir)
+            .map_err(|err| format!("Unable to open checkpoint file: {}", err))?;
+
+        Ok(journald_source(journal, out, checkpointer, units))
     }
 
     fn output_type(&self) -> DataType {
         DataType::Log
     }
 }
 
-fn journald_source<J>(journal: J, out: mpsc::Sender<Event>, units: HashSet<String>) -> super::Source
+fn journald_source<J>(
+    journal: J,
+    out: mpsc::Sender<Event>,
+    checkpointer: Checkpointer,
+    units: HashSet<String>,
+) -> super::Source
 where
-    J: 'static + Iterator<Item = Result<Record, Error>> + Send,
+    J: 'static + Iterator<Item = Result<Record, Error>> + JournalCursor + Send,
```

Comment:
minor nit: can you move the life time-bound to be the last bound?

Reply:
Done